mmengine/dataset/dataset_wrapper.py

```diff
@@ -45,13 +48,34 @@ def __init__(self,
                 raise TypeError(
                     'elements in datasets sequence should be config or '
                     f'`BaseDataset` instance, but got {type(dataset)}')
+        if ignore_keys is None:
+            self.ignore_keys = []
+        elif isinstance(ignore_keys, str):
+            self.ignore_keys = [ignore_keys]
+        elif isinstance(ignore_keys, list):
+            self.ignore_keys = ignore_keys
+        else:
+            raise TypeError('ignore_keys should be a list or str, '
+                            f'but got {type(ignore_keys)}')
+
+        meta_keys: set = set()
+        for dataset in self.datasets:
+            meta_keys |= dataset.metainfo.keys()
         # Only use metainfo of first dataset.
         self._metainfo = self.datasets[0].metainfo
         for i, dataset in enumerate(self.datasets, 1):
-            if self._metainfo != dataset.metainfo:
-                raise ValueError(
-                    f'The meta information of the {i}-th dataset does not '
-                    'match meta information of the first dataset')
+            for key in meta_keys:
+                if key in self.ignore_keys:
+                    continue
+                if key not in self._metainfo:
+                    raise ValueError(
+                        f'{key} does not in the meta information of '
+                        'the first dataset')
+                if key not in dataset.metainfo or \
+                        self._metainfo[key] != dataset.metainfo[key]:
```

Comment:
```suggestion
                self._metainfo[key] != dataset.metainfo[key]:
```

Reply:
if only use `self._metainfo[key] != dataset.metainfo[key]` it may get an error if `key not in dataset.metainfo`?